### PR TITLE
Add 'No native classes names in identifiers names' rule

### LINKED ISSUE #28: No native classes names in identifiers names
Forbid class names in variables. eg `varNameArray`, `varNameObject`. Variables should be named `varNames` if it is a collection of items.

### PR DESCRIPTION
**Pull request by PyTechMike**
Resolve #28

### CODE REVIEW

File: test/no-native-classes-names-in-identifiers-names.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		},
+		{
+			code: 'let testName = {}'
+		},
+		{
+			code: 'let testName = "string"'
+		},
+		{
+			code: 'let testName = true'
+		}
```
[the remark this answers is not in the document]
Comment (constgen) [reply test/no-native-classes-names-in-identifiers-names.test.js]: Add case `let numbers = [1,2,3]`

[the remark this answers is not in the document]
Comment (constgen) [reply test/no-native-classes-names-in-identifiers-names.test.js]: Add a case for an empty code

**Review by constgen — CHANGES_REQUESTED** [review]

File: src/rules/no-native-classes-names-in-identifiers-names.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+	meta: {
+		messages: {
+			noNativeClassesNamesInIdentifiersNames: 'Avoid using native classes names in identifiers names.'
```
[the remark this answers is not in the document]
Comment (PyTechMike) [reply src/rules/no-native-classes-names-in-identifiers-names.js]: Is this variant better? `Avoid using native classes names in identifiers (such as "Object" or "String")`

File: test/no-native-classes-names-in-identifiers-names.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (PyTechMike) [reply test/no-native-classes-names-in-identifiers-names.test.js]: > Add cases for reading identifiers with classes names. Such cases should not fail. Existing cases make no sense

What do you mean when you talk about "reading identifiers with classes names"? Can you add some examples? And why existing ones are senseless?

[the remark this answers is not in the document]
Comment (PyTechMike) [reply test/no-native-classes-names-in-identifiers-names.test.js]: > Add case `let numbers = [1,2,3]`

Is it going to fail or no?

[the remark this answers is not in the document]
Comment (PyTechMike) [reply test/no-native-classes-names-in-identifiers-names.test.js]: Is native class name could be only in the end of the identifier on in the middle/start also?

[the remark this answers is not in the document]
Comment (constgen) [reply test/no-native-classes-names-in-identifiers-names.test.js]: All this conversation is related to `valid` cases. Identifiers that doesn't contain classes names are not helpful. You may leave only one to be sure that you don't invalidate everything.

File: src/rules/no-native-classes-names-in-identifiers-names.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (constgen) [reply src/rules/no-native-classes-names-in-identifiers-names.js]: It is better. And can be used. But think what if you show what words exactly were used

File: src/rules/no-native-classes-names-in-identifiers-names.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (constgen) [reply src/rules/no-native-classes-names-in-identifiers-names.js]: `no-native-names`

File: src/rules/no-native-classes-names-in-identifiers-names.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+			noNativeClassesNamesInIdentifiersNames: 'Avoid using native classes names in identifiers names.'
+		}
+	},
+	create (context) {
+		return {
+			Identifier (node) {
+				const nativeClassNamesRegExp = /Number|number|String|string|Object|object|Function|function|Boolean|boolean|Bool|bool|Array|array/;
+				let nativeClassNameInIdentifier = nativeClassNamesRegExp.test(node.name);
+
+				if (nativeClassNameInIdentifier) {
```
[the remark this answers is not in the document]
Comment (constgen) [reply src/rules/no-native-classes-names-in-identifiers-names.js]: `isNativeClassNameInIdentifier` sounds like a function name

**Review by constgen — CHANGES_REQUESTED** [review]
Conversations updated

File: src/rules/no-native-classes-names-in-identifiers-names.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (PyTechMike) [reply src/rules/no-native-classes-names-in-identifiers-names.js]: > `isNativeClassNameInIdentifier` sounds like a function name

As I know, `is` prefix is compatible both for functions and booleans and the best way to name them

**Comment by PyTechMike** [comment]
Improve cases validation with natives in the end of identifiers

**Review by constgen — CHANGES_REQUESTED** [review]
Some comments are not resolved

**Comment by PyTechMike** [comment]
@constgen please check out my comment 
https://github.com/atomspace/eslint-plugin-naming-convention/issues/28#issuecomment-561339815

**Review by constgen — DISMISSED** [review]

**Review by constgen — APPROVED** [review]

**Pull request merged**